File: src/main/java/no/fint/featuretoggle/FintUnleashConfig.java
package no.fint.featuretoggle;

import no.finn.unleash.DefaultUnleash;
import no.finn.unleash.util.UnleashConfig;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class FintUnleashConfig {

    @Value("${spring.application.name}")
    private String applicationName;

    @Value("${fint.unleash.api:https://unleashed-beta.fintlabs.no/api/}")
    private String unleashApi;

    @Value("${fint.unleash.instance-id:backend}")
    private String instanceId;

    @Bean
    public DefaultUnleash unleash() {
        UnleashConfig config = UnleashConfig.builder()
                .appName(applicationName)
                .instanceId(instanceId)
                .unleashAPI(unleashApi)
                .build();

        return new DefaultUnleash(config);
    }
}

File: src/main/kotlin/no/fint/featuretoggle/config/FintUnleashConfig.kt
package no.fint.featuretoggle.config

import io.getunleash.DefaultUnleash
import io.getunleash.Unleash
import io.getunleash.util.UnleashConfig
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration

@ComponentScan(basePackageClasses = [UnleashProperties::class])
@Configuration
class FintUnleashConfig (
        @Value("\${spring.application.name}")
        private val applicationName: String,
        private val unleashProperties: UnleashProperties
) {
    @Bean
    fun unleashConfig(): UnleashConfig {
        return UnleashConfig.builder()
                .appName(applicationName)
                .instanceId(unleashProperties.instanceId)
                .unleashAPI(unleashProperties.api)
                .apiKey(unleashProperties.apiKey)
                .synchronousFetchOnInitialisation(true)
                .build()
    }

    @Bean
    fun unleashClient(unleashConfig: UnleashConfig): Unleash {
        return DefaultUnleash(unleashConfig)
    }
}